isobusfs/isobusfs_srv_fa.c

```diff
@@ -0,0 +1,923 @@
+		goto send_response;
+	}
+
+	if (handle->dir) {
+		error_code = isobusfs_srv_seek_directory(handle,
+							 le32toh(req->offset));
+		res.position = htole32(handle->dir_pos);
+	} else {
+		error_code = isobusfs_srv_seek(priv, handle, le32toh(req->offset),
+					       req->position_mode);
+		res.position = htole32(handle->offset);
+	}
+
+send_response:
+	res.fs_function =
+		isobusfs_cg_function_to_buf(ISOBUSFS_CG_FILE_ACCESS,
+					    ISOBUSFS_FA_F_SEEK_FILE_RES);
+	res.tan = req->tan;
+	res.error_code = error_code;
+
+	/* send to socket */
+	ret = isobusfs_srv_sendto(priv, msg, &res, sizeof(res));
+	if (ret < 0) {
+		pr_warn("can't send seek file response");
+		return ret;
+	}
+
+	pr_debug("> tx: Seek File Response. Error code: %d, offset: %d",
+		 error_code, offset_out);
+
+	return 0;
+}
+
+static int isobusfs_srv_fa_cf_req(struct isobusfs_srv_priv *priv,
+				  struct isobusfs_msg *msg)
+{
+	struct isobusfs_close_file_request *req;
+	struct isobusfs_close_file_res res;
+	struct isobusfs_srv_client *client;
+	uint8_t error_code = 0;
+	int ret;
+
+	req = (struct isobusfs_close_file_request *)msg->buf;
+
+	client = isobusfs_srv_get_client_by_msg(priv, msg);
+	if (!client) {
+		pr_warn("client not found");
+		error_code = ISOBUSFS_ERR_OTHER;
+		goto send_response;
+	}
+
+	ret = isobusfs_srv_release_handle(priv, client, req->handle);
+	if (ret < 0) {
+		pr_warn("failed to release handle: %x", req->handle);
+		switch (ret) {
+		case -ENOENT:
+			error_code = ISOBUSFS_ERR_FILE_ORPATH_NOT_FOUND;
+			break;
+		default:
+			error_code = ISOBUSFS_ERR_OTHER;
+		}
+	}
+
+send_response:
+	res.fs_function =
+		isobusfs_cg_function_to_buf(ISOBUSFS_CG_FILE_ACCESS,
+					    ISOBUSFS_FA_F_CLOSE_FILE_RES);
+	res.tan = req->tan;
+	res.error_code = error_code;
+	memset(&res.reserved[0], 0xff, sizeof(res.reserved));
+
+	/* send to socket */
+	ret = isobusfs_srv_sendto(priv, msg, &res, sizeof(res));
+	if (ret < 0) {
+		pr_warn("can't send current directory response");
+		goto err;
+	}
+
+	pr_debug("> tx: Close File Response. Error code: %d", error_code);
+
+err:
+	return ret;
+}
+
+/* Command group: file access */
+int isobusfs_srv_rx_cg_fa(struct isobusfs_srv_priv *priv,
+			  struct isobusfs_msg *msg)
+{
+	int func = isobusfs_buf_to_function(msg->buf);
+	int ret = 0;
+
+	switch (func) {
+	case ISOBUSFS_FA_F_OPEN_FILE_REQ:
+		ret = isobusfs_srv_fa_open_file_req(priv, msg);
+		break;
+	case ISOBUSFS_FA_F_CLOSE_FILE_REQ:
+		ret = isobusfs_srv_fa_cf_req(priv, msg);
+		break;
+	case ISOBUSFS_FA_F_READ_FILE_REQ:
+		ret = isobusfs_srv_fa_rf_req(priv, msg);
+		break;
+	case ISOBUSFS_FA_F_SEEK_FILE_REQ:
+		ret = isobusfs_srv_fa_sf_req(priv, msg);
+		break;
+	case ISOBUSFS_FA_F_WRITE_FILE_REQ: /* fall through */
+	default:
+		goto not_supported;
+	}
+
+	return ret;
+
+not_supported:
+	isobusfs_srv_send_error(priv, msg, ISOBUSFS_ERR_FUNC_NOT_SUPPORTED);
+
+	pr_warn("%s: unsupported function: %i", __func__, func);
+
+	/* Not a critical error */
+	return 0;
+}
```

Comment:
Simplify as:
```suggestion
	default:
		isobusfs_srv_send_error(priv, msg, ISOBUSFS_ERR_FUNC_NOT_SUPPORTED);

		pr_warn("%s: unsupported function: %i", __func__, func);

		/* Not a critical error */
	}

	return 0;
}
```
then remove `ret` declaration.